sql/sqlbase/metadata.go

```diff
@@ -94,12 +94,6 @@ func (ms *MetadataSchema) AddDescriptor(parentID ID, desc DescriptorProto) {
 	if id := desc.GetID(); id > keys.MaxReservedDescID {
 		panic(fmt.Sprintf("invalid reserved table ID: %d > %d", id, keys.MaxReservedDescID))
 	}
-	for _, d := range ms.descs {
-		if d.desc.GetID() == desc.GetID() {
-			log.Errorf(context.TODO(), "adding descriptor with duplicate ID: %v", desc)
```

Comment:
I removed it because it was put into place recently to accommodate test1 and test2. I've  reverted it back because I agree with you that it's better to be safe than sorry.